web/htdocs/assets/js/codesearch_ui.js

```diff
@@ -272,6 +272,16 @@ var FileMatch = Backbone.Model.extend({
   },
 
   url: function() {
```

Comment:
The code duplication with Match, above, is unfortunate.

Any way to combine those nicely?

Reply:
OK, I've pulled that logic out into some shared functions. (If I'd done that the first time, I probably could have avoided this bug in the first place!)